Assignment_2-data-emma-luk.js

```diff
@@ -0,0 +1,101 @@
+﻿// Emma Luk
+// Assignments 2 Using the revealing module pattern, build a Date utility module 
+
+var myDate = (function iife() {
+    var displaysModuleDate = null;
+
+    //private function: date instantiation
+    function _initDate() {
+        return new Date();
+    }
+
+    //check here that date is set: if not, initialize date to now
+    function setDate(date) {
+        if (date === undefined && null) {
+            displaysModuleDate = _initDate();
+        }
+        else if (typeof date === 'number' && !isNaN(date)) {
+            displaysModuleDate = new Date(date);
+        }
+        else if (typeof date === 'object') {
+            displaysModuleDate = date;
+        }
+        else {
+            console.log("sorry!! wrong type");
+        }
+    }
+
+    function getDate(objFormat) {
+        if (displaysModuleDate === null && undefined) {
+            displaysModuleDate = _initDate();
+        }
+        if (objFormat.format === "formatted") {
+            return displaysModuleDate.getMonth() + " " + displaysModuleDate.getDate() + ", " + displaysModuleDate.getFullYear();
+        }
+        else if (objFormat.format === "milliseconds" || objFormat === undefined) {
+            return displaysModuleDate.getDate();
+        }
+        else {
+            console.log("sorry!! unknown format");
+        }
+    }
+
+    function getDayName() {
+        var day = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];

```

Comment:
This is a great way to implement this, but put your array outside of this function, but still inside the module. Every time this method is called it's going to re-declare and reserve memory for this array which will never change. It would be more efficient to declare it outside of this method.